File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/generator/ClassGeneratorInterface.java
package com.github.tobi812.sprykerplugin.models.generator;

import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;

import java.util.HashMap;

public interface ClassGeneratorInterface {
    PhpClassInterface generateClass(String classType, HashMap<String, String> config) throws Exception;
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/generator/ClassGeneratorInterface.kt
package com.github.tobi812.sprykerplugin.models.generator

import com.github.tobi812.sprykerplugin.actions.ClassConfig
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface
import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import java.lang.Exception

interface ClassGeneratorInterface {
    @Throws(Exception::class)
    fun generateClass(classType: String, config: ClassConfig): PhpClassInterface

    fun createFullQualifiedName(classDefinition: ClassDefinitionInterface, config: ClassConfig): String
}
